### ISSUE TITLE
Using Ctrl-s and Restart Leo?, Leo Forgets To Reopen Outlines

### ISSUE THREAD

**Issue by LewisNeal** [leo-editor/leo-editor#2433]
If I use Ctrl-s to save a leo file I get message
"g:/my drive/editor/workbook.leo has changed outside Leo. Restart Leo? Yes/No"

press Yes and Leo then opens a workbook file in my home directory - a different path 
If other files were open when Leo closed they do not reopen. Similar behaviour in #2412 

_Leo Log Window
Leo 6.6b2-devel, devel branch, build ae2cf5da30
2022-02-18 17:12:15 -0600
Python 3.10.1, PyQt version 6.2.3
Windows 10 AMD64 (build 10.0.19042) SP0_

**Comment by tbpassin** [comment]
in SessionManager.save_snapshot(), when closed with the window's close box, the session contains the open windows.  When closing with CTRL-q, the session list is empty.  Wierd!

**Comment by tbpassin** [comment]
In SessionManager.get_session(), which is called during the exit command, if Leo is closed with CTRL-q, get_session() is called **twice**.  The second time leads to the empty sesssion list of windows, which is then saved, overwriting the previous correct session list.

When Leo is closed by clicking on the window's close box, get_session() is called only once.

**Comment by tbpassin** [comment]
Running Leo with --trace=shutdown, the trace outputs a list of calls ending with

```
destroyAllOpenWithFiles
LeoQtGui.destroySelf: calling qtApp.Quit
```
That's with using the close box.  However, using CTRL-q, after these two calls,  `get_session()` is called **again**.

I'm stuck here for now.  I can't see how this happens.

**Comment by tbpassin** [comment]
I found a _qt_gui.close_event_, which also calls `save_snapshot()`. Maybe there is a hook that calls _qt_gui.close_event_ on closing, which might account for the double call to `get_session()`.

**Comment by tbpassin** [comment]
When the window's close box is clicked, the _exit-leo_  command is not called.  When Leo is closed using CTRL-q, the command is called.  It seems strange that Leo would shut down differently.

**Comment by tbpassin** [comment]
So in the end, the solution is simple.  First, add the import of _QCloseEvent_ to leoQt5 and leoQt6. Import it in the _imports_ section of leoApp.py. Then for the _exit-leo_ command in leoApp.py, the entire command becomes:
```
@cmd('exit-leo')
@cmd('quit-leo')
def onQuit(self, event=None):
    """Exit Leo, prompting to save unsaved outlines first."""
    if 'shutdown' in g.app.debug:
        g.trace()
    # #2433 - use the same method as clicking on the close box.
    g.app.gui.close_event(QCloseEvent())
```
  

Doing it this way also lets us avoid having to figure out the problem with the existing code for this command.  If @Edward agrees with this solution, I'll put in a PR.

**Comment by edreamleo** [comment]
@tbpassin Excellent detective work. Please file a PR. Thanks!

**Comment by edreamleo** [comment]
Fix by PR #2434.
